app/routes/__auth/user.server.ts

```diff
@@ -14,8 +16,55 @@ export async function getUser({ headers }: Request) {
   const email = session.get('email') as string
   const groups = session.get('groups') as string[]
   const idp = session.get('idp') as string | null
-
+  const refreshToken = session.get('refreshToken') as string
+  const cognitoUserName = session.get('cognitoUserName') as string
+  const token = session.get('token') as string
   if (!sub) return null
+  const user = { sub, email, groups, idp, refreshToken, cognitoUserName, token }
+  if (!token) {
```

Comment:
or if token has expired...

Reply:
Added a check for expiration in latest commit